code/modules/projectiles/projectile_base.dm

```diff
@@ -400,11 +460,78 @@
 	if(QDELETED(src)) // we coulda bumped something
 		return
 	if(trajectory && !trajectory_ignore_forcemove && isturf(target))
+		if(hitscan)
+			finalize_hitscan_and_generate_tracers(FALSE)
 		trajectory.initialize_location(target.x, target.y, target.z, 0, 0)
+		if(hitscan)
+			record_hitscan_start(RETURN_PRECISE_POINT(src))
 
 /obj/item/projectile/proc/is_reflectable(desired_reflectability_level)
 	if(reflectability == REFLECTABILITY_NEVER) //You'd trust coders not to try and override never reflectable things, but heaven help us I do not
 		return FALSE
 	if(reflectability < desired_reflectability_level)
 		return FALSE
 	return TRUE
+
+/obj/item/projectile/proc/record_hitscan_start(datum/point_precise/point_cache)
+	if(point_cache)
+		beam_segments = list()
+		beam_index = point_cache
+		beam_segments[beam_index] = null //record start.
+
+/obj/item/projectile/proc/process_hitscan()
+	var/safety = range * 10
```

Comment:
A comment here explaining exactly what safety represents here would be good

Reply:
Done